### PR TITLE
[pytorch] [distributed] [easy] Added test case for reinit

### PR DESCRIPTION
**Pull request by satgera**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26506 [pytorch] [distributed] [easy] Added test case for reinit**

[pytorch] [distributed] Made test forgiving to allow rpc agent to return one of the two errors.

Differential Revision: [D17488899](https://our.internmc.facebook.com/intern/diff/D17488899/)

### CODE REVIEW

File: test/test_rpc.py [hunk 1]
```diff
@@ -133,7 +133,12 @@ def test_reinit(self):
         store = dist.FileStore(self.file.name, self.world_size)
         dist.init_process_group(backend="gloo", rank=self.rank,
                                 world_size=self.world_size, store=store)
-        with self.assertRaisesRegex(RuntimeError, "is not unique"):
+        dist.init_model_parallel(self_name='worker{}'.format(self.rank),
```
Comment (pritamdamania87) [inline test/test_rpc.py]: Shouldn't we have two separate tests here? One uses duplicate worker name and raises an error with message "is not unique" and then another initializes model_parallel again with a unique name but should raise an error with "is already initialized"?

**Review by pritamdamania87 — CHANGES_REQUESTED** [review]

File: test/test_rpc.py [hunk 2]
```diff
@@ -129,17 +129,42 @@ def test_self_add(self):
...
+                                 backend=BACKEND,
+                                 self_rank=self.rank,
+                                 init_method=RPC_INIT_URL)
+        with self.assertRaisesRegex(RuntimeError,
+                                    "(is not unique|is already initialized)"):
             dist.init_model_parallel(self_name="duplicate_name",
                                      backend=BACKEND,
                                      self_rank=self.rank,
                                      init_method=RPC_INIT_URL)
```
Comment (pietern) [inline test/test_rpc.py]: AFAICT this shouldn't initialize with `init_model_parallel` outside the `assertRaisesRegex` wrapper.

The purpose is to only test for duplicate names right? Not that it was already initialized.

**Review by pietern — APPROVED** [review]
LGTM, provided CI passes.

**Comment by satgera** [comment]
Retried the CI to ensure all pass but looks like infra issue. Some of the failure which indicate build failure seems to have succeed.
Since this is a trivial change and other test have passed, landing.

**Pull request closed without merging**